src/compass/utils/ale.py

```diff
@@ -0,0 +1,536 @@
+    upperleft_x = int(np.round(x_loc)) - margin // 2
+    upperleft_y = int(np.round(y_loc)) - margin // 2
+    lowerright_x = upperleft_x + margin
+    lowerright_y = upperleft_y + margin
+
+    if (upperleft_x < 0) or (upperleft_y < 0) or \
+            (lowerright_x > arr.shape[1]) or (lowerright_y > arr.shape[0]):
+        err_msg = 'The corner reflector input coordinates are outside of the CSLC'
+        raise ValueError(err_msg)
+
+    # Extract an area around x_loc, y_loc
+    img = arr[upperleft_y:lowerright_y, upperleft_x:lowerright_x]
+
+    # Oversample CSLC subset and get amplitude
+    img_ovs = isce3.cal.point_target_info.oversample(
+        img, ovs_factor)
+    idx_peak_ovs = np.argmax(np.abs(img_ovs))
+    img_peak_ovs = np.unravel_index(idx_peak_ovs, img_ovs.shape)
+
+    x_chip = x_vect[0] + x_spac * upperleft_x
+    y_chip = y_vect[0] + y_spac * upperleft_y
+
+    dx1 = x_spac / ovs_factor
+    dy1 = y_spac / ovs_factor
+
+    x_cr = x_chip + x_spac / 2 + img_peak_ovs[1] * dx1
+    y_cr = y_chip + y_spac / 2 + img_peak_ovs[0] * dy1
+
+    return x_cr, y_cr
+
+
+def get_cslc(cslc_file, mission_id='S1', pol='VV'):
+    '''
+    Get CSLC-S1 array associated to 'pol'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+    mission_id: str
+        Mission identifier. S1: Sentinel-1 or
+        NI: NISAR (default: S1)
+    pol: str
+        Polarization string of CSLC-S1 array
+        to extract from product
+
+    Returns
+    -------
+    cslc: np.ndarray
+        Geocoded SLC image corresponding to 'pol'
+        polarization channel.
+    '''
+
+    if mission_id == 'S1':
+        cslc_path = f'science/SENTINEL1/CSLC/grids/{pol}'
+    elif mission_id == 'NI':
+        cslc_path = f'/science/LSAR/GSLC/grids/frequencyA/{pol}'
+    else:
+        err_str = f'{mission_id} is not a valid mission identifier'
+        raise ValueError(err_str)
+
+    with h5py.File(cslc_file, 'r') as h5:
+        cslc = h5[cslc_path][()]
+
+    return cslc
+
+
+def get_xy_info(cslc_file, mission_id='S1', pol='VV'):
+    '''
+    Get X/Y spacings and coordinate vectors from the
+    geocoded SLC contained in 'cslc_file'
+
+    Parameters
+    ----------
+    cslc_file: str
+        File path to CSLC-S1 product
+    mission_id: str
+        Mission identifier. S1: Sentinel-1 or
+        NI: NISAR (default: S1)
+    pol: str
+        Polarization channel of the geocoded SLC
+        to analyze (default: VV)
+
+    Returns
+    -------
+    x_vect: np.ndarray
+        Array of X-coordinates associated to CSLC-S1
+    y_vect: np.ndarray
+        Array of Y-coordinates associated to CSLC-S1
+    x_spac: np.float
+        CSLC-S1 spacing along X-direction
+    y_spac: np.float
+        CSLC-S1 spacing along Y-direction
+    '''
+    if mission_id == 'S1':
+        cslc_path = '/science/SENTINEL1/CSLC/grids/'
+    elif mission_id == 'NI':
+        cslc_path = '/science/LSAR/GSLC/grids/'
```

Comment:
Need to pass frequency for NISAR
```suggestion
        cslc_path = '/science/LSAR/GSLC/grids/frequency'
```